### PR TITLE
Modify dictionary search

### PR DESCRIPTION
**Pull request by khoidt**

### CODE REVIEW

File: ebl/dictionary/infrastructure/word_repository.py [hunk 1]
```diff
@@ -72,6 +72,51 @@ def _create_lemma_search_pipeline(query):
     ]
 
 
+def _create_query_by_lemma(word: str, collate: bool) -> dict:
+    regex_options = "mi" if collate else "m"
+    return {
```
Comment (khoidt) [inline ebl/dictionary/infrastructure/word_repository.py:77]: Check if "m" is needed

File: ebl/tests/dictionary/test_dictionary.py [hunk 2]
```diff
@@ -46,8 +48,9 @@ def test_search_finds_all_homonyms(dictionary, word):
     another_word = {**word, "_id": "part1 part2 II", "homonym": "II"}
     dictionary.create(word)
     dictionary.create(another_word)
+    query = urlencode({"word": word["lemma"][0]}, doseq=False)
```
Comment (khoidt) [inline ebl/tests/dictionary/test_dictionary.py:51]: Check if `doseq` is needed

File: ebl/tests/dictionary/test_dictionary.py [hunk 3]
```diff
@@ -59,20 +62,64 @@ def test_search_finds_by_meaning(dictionary, word):
...
     dictionary.create(word)
     dictionary.create(another_word)
+    query = urlencode({"meaning": word["meaning"]}, doseq=False)
 
-    assert dictionary.search(word["meaning"][1:4]) == [word]
+    assert dictionary.search(query) == [word]
+
+
+def test_search_finds_by_root(dictionary, word):
+    another_word = copy.deepcopy({**word, "_id": "part1 part2 II", "homonym": "II"})
```
Comment (khoidt) [inline ebl/tests/dictionary/test_dictionary.py:71]: Check if `copy.copy()` works

Comment (khoidt) [inline ebl/tests/dictionary/test_dictionary.py:71]: Remove `copy.deepcopy()`

**Review by fsimonjetz — APPROVED** [review]

**Pull request merged**